### PR TITLE
refactor/480

### LINKED ISSUE #480: Consolidate warnings for missing fixes in airport files
The sim currently puts warnings into the console for any fixes that are being used but have not been defined in the `fixes` section of the airport json file. They show up as separate entries, and often multiple times. If they could instead be put into an array that clears duplicate elements and sorts, that would be a much easier thing to work with. For example, I copy and paste that list into skyvector (since its waypoint database seems to be the most comprehensive), use the nav log thing to generate a pdf including the coordinates, and then sublime to reformat them as needed.

image

Related: #94.

### PR DESCRIPTION
**Pull request by erikquinn**
Resolves #480.

Consolidate console warnings for missing fixes to single message

### CODE REVIEW

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js [hunk 1]
```diff
@@ -316,4 +322,21 @@ export default class NavigationLibrary {
             flightPhase === FLIGHT_PHASE.TAXI ||
             flightPhase === FLIGHT_PHASE.WAITING;
     }
+
+    showConsoleWarningForUndefinedFixes() {
+        const allFixNames = this._getAllFixNames();
+        const missingFixes = allFixNames.filter((fix) => !FixCollection.findFixByName(fix));
+
+        console.warn(`The following fixes have yet to be defined in the "fixes" section: ${missingFixes}`);
```
Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: `if (missingFixes.length > 0)`

Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: instead `if (missingFixes.length < 1) {return;}`.

done

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js [hunk 2]
```diff
@@ -316,4 +322,41 @@ export default class NavigationLibrary {
...
+
+    /**
+     * Gathers a unique, sorted list of all fixes used in all known procedures
+     *
+     * @for NavigationLibrary
+     * @method _getAllFixNames
+     * @return {array<string>} ['fixxa', 'fixxb', 'fixxc', ...]
+     * @private
+     */
+    _getAllFixNames() {
```
Comment (n8rzz) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: Could we call this something like `_getAllFixNamesInUse()`?  Or somehow make me t clear that these fixes are used?

File: src/assets/scripts/client/navigationLibrary/StandardRoute/RouteSegmentCollection.js [hunk 3]
```diff
@@ -137,6 +138,23 @@ export default class RouteSegmentCollection extends BaseCollection {
...
+     * Returns an array of fix names used by any segment in the collection
+     *
+     * @for RouteSegmentCollection
+     * @method getAllFixNames
+     * @return {array<string>} ['fixname', 'fixname', 'fixname', ...]
+     */
+    getAllFixNames() {
+        let fixNames = [];
+
+        this._items.forEach((routeSegmentModel) => {
```
Comment (n8rzz) [inline src/assets/scripts/client/navigationLibrary/StandardRoute/RouteSegmentCollection.js]: Why not use `_map()` here and set the result to `fixNames`

File: src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteCollection.js [hunk 4]
```diff
@@ -212,6 +212,23 @@ export default class StandardRouteCollection extends BaseCollection {
...
+     *
+     * @for StandardRouteCollection
+     * @method getAllFixNames
+     * @return {array<string>} ['fixname', 'fixname', 'fixname', ...]
+     */
+    getAllFixNames() {
+        let fixNames = [];
+
+        this._items.forEach((standardRouteModel) => {
+            fixNames = fixNames.concat(standardRouteModel.getAllFixNames());
```
Comment (n8rzz) [inline src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteCollection.js]: Why not use `_map()` here and set the result to `fixNames`.  Then youcan use `const` as well.

File: src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteModel.js [hunk 5]
```diff
@@ -291,6 +291,24 @@ export default class StandardRouteModel extends BaseModel {
     }
 
     /**
+     * Returns an array of fix names used by any portion of the procedure
+     *
+     * @for StandardRouteModel
+     * @method getAllFixNames
+     * @return {array<string>} ['fixname', 'fixname', 'fixname', ...]
+     */
+    getAllFixNames() {
```
Comment (n8rzz) [inline src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteModel.js]: I have thoughts on this. It I need a computer to show you. In the mean time, why not `_flatten()` for the draw section?

File: test/navigationLibrary/StandardRoute/RouteSegmentCollection.spec.js [hunk 6]
```diff
@@ -53,13 +54,21 @@ ava('.findWaypointsForSegmentName() returns an array of fixes for a given segmen
...
+ava('.gatherFixNamesForCollection() returns a list of entryPoint/exitPoint names for each RouteSegmentModel in the collection', t => {
     const collection = new RouteSegmentCollection(ROUTE_SEGMENTS_MOCK);
     const result = collection.gatherFixNamesForCollection();
 
     t.true(result.length === 8);
 });
 
+ava('.getAllFixNames() returns an array of fix names used by any segment in the collection', (t) => {
+    const collection = new RouteSegmentCollection(ROUTE_SEGMENTS_MOCK);
+    const result = _uniq(collection.getAllFixNames().sort());
```
Comment (n8rzz) [inline test/navigationLibrary/StandardRoute/RouteSegmentCollection.spec.js]: What's the need for _uniq here?

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js [hunk 7]
```diff
@@ -316,4 +322,41 @@ export default class NavigationLibrary {
...
+        const sidFixes = this.sidCollection.getAllFixNames();
```
Comment (n8rzz) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: There are a few blocks below with this same structure, and my suggestion here will apply to those as well.

Instead of `_uniq(thing.concat(otherThing)).sort`, we can do this:
```javascript
const allFixNames = [
  ...this.sidCollection.getAllFixNames(),
  ...this.starCollection.getAllFixNames()
];
const uniqueFixNames = _uniq(allFixNames);
const allNonVectorFixes = _filter(uniqueFixNames, (fix) => fix.indexOf(VECTOR_WAYPOINT_PREFIX) === INVALID_INDEX);

return allNonVectorFixes.sort();
```
This does a few things for us:
- gets rid of `concat` which will mutate an existing array (not desirable).
- gives us an aptly named var `uniqueFixNames` that provides a little more context
- adds `_filter()` which, although there is a native version, we are using lodash all over the place and should use lodash methods when available.  Plus native `filter` isn't supported by some versions of IE (though this argument is not a valid today as it was a year or two ago)

File: src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteModel.js (hunk 5 above)
Comment (n8rzz) [inline src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteModel.js]: see above massive comment that also applies here

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js (hunk 2 above)
Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: yeah, that works.

done

File: src/assets/scripts/client/navigationLibrary/StandardRoute/RouteSegmentCollection.js (hunk 3 above)
Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/StandardRoute/RouteSegmentCollection.js]: Because with map I'll end up with nested arrays rather than a big pile of fixes. I could `.map()` and then `_flatten`?

File: src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteCollection.js (hunk 4 above)
Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteCollection.js]: discussed above

File: test/navigationLibrary/StandardRoute/RouteSegmentCollection.spec.js (hunk 6 above)
Comment (erikquinn) [inline test/navigationLibrary/StandardRoute/RouteSegmentCollection.spec.js]: Just so I could explicitly write out the full list of fixes collected at this point in the following line, and ensure that was the output. I could extend that `expectedResult` to contain all of the duplicated fixes, but that seemed clunky (not that this is a real great answer either if you ask me).

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js (hunk 7 above)
Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: love it!

done

Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: Other than `_flatten`, I can't figure out how to redo the other methods. They are different because the spread is not done within an arrow function (where they seem to be illegal)?

Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: Oh wait, I think this can be done with reduce. :thinking:

Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: Sort of. I could do the following, which maybe is better:
```
// BEFORE
getAllFixNamesInUse() {
    let fixNames = [];

    this._items.forEach((routeSegmentModel) => {
        fixNames = fixNames.concat(routeSegmentModel.getAllFixNamesInUse());
    });

    return fixNames;
}

// AFTER
getAllFixNamesInUse() {
    const allFixNames = this._items.map((routeSegmentModel) => routeSegmentModel.getAllFixNamesInUse())
        .reduce((fixList, fixGroup) => fixList.concat(fixGroup));

    return allFixNames;
}
```

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js [hunk 8]
```diff
@@ -316,4 +323,45 @@ export default class NavigationLibrary {
...
+     * @private
+     */
+    _getAllFixNamesInUse() {
+        const allFixNames = [
+            ...this.sidCollection.getAllFixNamesInUse(),
+            ...this.starCollection.getAllFixNamesInUse()
+        ];
+        const uniqueFixNames = _uniq(allFixNames);
+        const allNonVectorFixes = _filter(uniqueFixNames, (fix) => {
+            return fix.indexOf(VECTOR_WAYPOINT_PREFIX) === INVALID_INDEX;
```
Comment (n8rzz) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: The return is unnecessary here, you can use a direct return without curly braces

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js (hunk 7 above)
Comment (n8rzz) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: You're still working too hard here. 

```javascript
const fixNames = _map(this._items, (routeSegmentModel) => {
  return [...routeSegmentModel.getAllFixNamesInUse()];
});
```

File: test/navigationLibrary/StandardRoute/RouteSegmentCollection.spec.js (hunk 6 above)
Comment (n8rzz) [inline test/navigationLibrary/StandardRoute/RouteSegmentCollection.spec.js]: I ask because you are checking the result based on the return of another function. In these instances, usually, I prefer to have an explicit list to compare against because that won't ever change.  When one of those methods changes and the test fails, you now have two vectors of failure instead of just one. 

I know where you're going here, and it's ok, I just want to call out that slight difference.

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js (hunk 7 above)
Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: That still yields a nested array though. There's no difference in result between `routeSegmentModel.getAllFixNamesInUse()` and `[...routeSegmentModel.getAllFixNamesInUse()]`. :confused:

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js (hunk 8 above)
Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: :+1: It was only to break the line for the line-length lint warning. Should I disable the warning for this line, or leave it as shown?

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js (hunk 7 above)
Comment (n8rzz) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: Hmm 🤔 

Not having a laptop is a real pain. Flatten may be the way to go then.

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js (hunk 8 above)
Comment (n8rzz) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: You can put the part after the arrow on the next line.

File: src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteCollection.js (hunk 4 above)
Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteCollection.js]: `_uniq(_flatten(_map))`

done

File: src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteModel.js (hunk 5 above)
Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteModel.js]: expanded into a new array with spread operators to combine everything, then uniq-ed it at the end.

done

File: test/navigationLibrary/StandardRoute/RouteSegmentCollection.spec.js (hunk 6 above)
Comment (erikquinn) [inline test/navigationLibrary/StandardRoute/RouteSegmentCollection.spec.js]: After thinking about it more, the function itself should be returning a unique array anyway, so amended it as such, and removed the uniq from the test, while leaving the unique `expectedResults` (effectively verifying it the same way).

done

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js (hunk 7 above)
Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: Very good. Used flatten to get it working.

done

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js (hunk 8 above)
Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: Would it be A or B?
```javascript
// A
const allNonVectorFixes = _filter(uniqueFixNames, (fix) =>
    fix.indexOf(VECTOR_WAYPOINT_PREFIX) === INVALID_INDEX);

// B
const allNonVectorFixes = _filter(uniqueFixNames, (fix) =>
    fix.indexOf(VECTOR_WAYPOINT_PREFIX) === INVALID_INDEX
);
```

Comment (n8rzz) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: @erikquinn `A`. option `B` has incorrect syntax.

I think you were getting at the semi=colon part of it, though.  there should be a semi-colon at the end (but also the missing parenthesis)

File: src/assets/scripts/client/navigationLibrary/StandardRoute/RouteSegmentCollection.js (hunk 3 above)
Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/StandardRoute/RouteSegmentCollection.js]: done

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js (hunk 8 above)
Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: they're both valid, but yes, that was the question. I'll move the `);` back onto the end of the same line.

File: src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteModel.js [hunk 9]
```diff
@@ -291,6 +291,25 @@ export default class StandardRouteModel extends BaseModel {
...
 
     /**
+     * Returns an array of unique fix names used by any portion of the procedure
+     *
+     * @for StandardRouteModel
+     * @method getAllFixNamesInUse
+     * @return {array<string>} ['fixname', 'fixname', 'fixname', ...]
+     */
+    getAllFixNamesInUse() {
+        const allFixNames = [
```
Comment (n8rzz) [inline src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteModel.js]: what happens if any one of these collections is `undefined`?

I think this would only be a problem during development of a new airport, but this may be an edge case worth writing a test for?

File: src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteWaypointModel.js [hunk 10]
```diff
@@ -324,10 +324,6 @@ export default class StandardRouteWaypointModel extends BaseModel {
         const fixModel = FixCollection.findFixByName(name);
 
         if (!fixModel) {
-            // TODO: This console warn should be done elsewehere, so a single console
-            // message can inform the airport designer of all missing fixes at once.
-            console.warn(`The following fix was not found in the list of fixes for this Airport: ${this.name}`);
-
             return this;
```
Comment (n8rzz) [inline src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteWaypointModel.js]: do we need to return `this` here or can this be a simple `return`?

File: test/navigationLibrary/StandardRoute/RouteSegmentModel.spec.js [hunk 11]
```diff
@@ -1,6 +1,8 @@
 /* eslint-disable arrow-parens, import/no-extraneous-dependencies*/
 import ava from 'ava';
 import sinon from 'sinon';
+import _isEqual from 'lodash/isEqual';
+import _uniq from 'lodash/uniq';
```
Comment (n8rzz) [inline test/navigationLibrary/StandardRoute/RouteSegmentModel.spec.js]: is `_uniq()` still in use within this file?

File: test/navigationLibrary/StandardRoute/StandardRouteModel.spec.js [hunk 12]
```diff
@@ -3,6 +3,7 @@ import sinon from 'sinon';
 import _isEqual from 'lodash/isEqual';
 import _keys from 'lodash/keys';
 import _map from 'lodash/map';
+import _uniq from 'lodash/uniq';
```
Comment (n8rzz) [inline test/navigationLibrary/StandardRoute/StandardRouteModel.spec.js]: is `_uniq()` still in use within this file?

File: test/navigationLibrary/StandardRoute/StandardRouteModel.spec.js [hunk 13]
```diff
@@ -3,6 +3,7 @@ import sinon from 'sinon';
...
 
 import StandardRouteModel from '../../../src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteModel';
```
Comment (n8rzz) [inline test/navigationLibrary/StandardRoute/StandardRouteModel.spec.js]: as mentioned above, we should add a test for cases where one of the collections is `undefined`.

for setup, you could do:
```javascript
const standardRouteModel = new StandardRouteModel(STAR_LIST_MOCK.GRNPA1);
standardRouteModel._exitCollection = undefined;
```
then run `standardRouteModel.getAllFixNamesInUse()` and see if stuff blows up.

File: src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteModel.js (hunk 9 above)
Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteModel.js]: well all of them will at least return an empty array if it all goes wrong, and they are allowed to have some of them empty anyway. So there should be no throw from trying to do `...undefined` or something because it will always at least be `...[]`, which resolves to `[]`.

File: src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteWaypointModel.js (hunk 10 above)
Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteWaypointModel.js]: There is no chaining done on this method anyway, so I can just have it return nothing when it's done.

File: test/navigationLibrary/StandardRoute/RouteSegmentModel.spec.js (hunk 11 above)
Comment (erikquinn) [inline test/navigationLibrary/StandardRoute/RouteSegmentModel.spec.js]: Yes, all imports are used.

File: test/navigationLibrary/StandardRoute/StandardRouteModel.spec.js (hunk 12 above)
Comment (erikquinn) [inline test/navigationLibrary/StandardRoute/StandardRouteModel.spec.js]: Yes, all imports are used.

File: test/navigationLibrary/StandardRoute/StandardRouteModel.spec.js (hunk 13 above)
Comment (erikquinn) [inline test/navigationLibrary/StandardRoute/StandardRouteModel.spec.js]: done

File: src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteModel.js (hunk 9 above)
Comment (erikquinn) [inline src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteModel.js]: added a test that ensures the correct content is returned without throwing even if any of the collections are empty.

done

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js [hunk 14]
```diff
@@ -289,12 +296,15 @@ export default class NavigationLibrary {
                 route = this._sidCollection.findRouteByIcao(routeString);
 
                 break;
-            case PROCEDURE_TYPE.STAR:
+
```
Comment (n8rzz) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: The empty line is unnecessary

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js [hunk 15]
```diff
@@ -289,12 +296,15 @@ export default class NavigationLibrary {
...
+            case PROCEDURE_TYPE.STAR: {
                 const { procedure } = new RouteModel(routeString);
 
                 route = this._starCollection.findRouteByIcao(procedure);
 
                 break;
+            }
+
```
Comment (n8rzz) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: The empty line is unnecessary

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js [hunk 16]
```diff
@@ -316,4 +326,44 @@ export default class NavigationLibrary {
...
+     * @private
+     */
+    _getAllFixNamesInUse() {
+        const allFixNames = [
+            ...this.sidCollection.getAllFixNamesInUse(),
+            ...this.starCollection.getAllFixNamesInUse()
+        ];
+        const uniqueFixNames = _uniq(allFixNames);
+        const allNonVectorFixes = _filter(uniqueFixNames, (fix) =>
+            fix.indexOf(VECTOR_WAYPOINT_PREFIX) === INVALID_INDEX);
```
Comment (n8rzz) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: Isn't there a method on the `RouteModel` that does this already?

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js [hunk 17]
```diff
@@ -316,4 +326,44 @@ export default class NavigationLibrary {
...
+     * @return {array<string>} ['fixxa', 'fixxb', 'fixxc', ...]
+     * @private
+     */
+    _getAllFixNamesInUse() {
+        const allFixNames = [
+            ...this.sidCollection.getAllFixNamesInUse(),
+            ...this.starCollection.getAllFixNamesInUse()
+        ];
+        const uniqueFixNames = _uniq(allFixNames);
+        const allNonVectorFixes = _filter(uniqueFixNames, (fix) =>
```
Comment (n8rzz) [inline src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: This is nit-picky,  it instead of `fix` can we use `fixName` here?  With simply `fix` someone could assume that is a reference to the `FixModel`.

File: src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteModel.js [hunk 18]
```diff
@@ -291,6 +291,25 @@ export default class StandardRouteModel extends BaseModel {
...
+            ...this._entryCollection.getAllFixNamesInUse(),
+            ...this._bodySegmentModel.getAllFixNamesInUse(),
+            ...this._exitCollection.getAllFixNamesInUse(),
+            ...this.draw.reduce((list, fix) => list.concat(fix)).map((fixName) => fixName.replace('*', ''))
```
Comment (n8rzz) [inline src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteModel.js]: There is a lot going on and n this one line.  This should be set to a var instead of inline like this.

File: src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteWaypointModel.js [hunk 19]
```diff
@@ -294,16 +294,12 @@ export default class StandardRouteWaypointModel extends BaseModel {
     }
 
     // TODO: why do we need to clone?
-    // TODO ClonePoisitonFromFix -> clonePositionFromFix
```
Comment (n8rzz) [inline src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteWaypointModel.js]: It could, but a little extra verbosity never hurts.

File: test/navigationLibrary/StandardRoute/RouteSegmentCollection.spec.js [hunk 20]
```diff
@@ -53,13 +54,21 @@ ava('.findWaypointsForSegmentName() returns an array of fixes for a given segmen
...
 
     t.true(result.length === 8);
 });
 
+ava('.getAllFixNamesInUse() returns an array of fix names used by any segment in the collection', (t) => {
+    const collection = new RouteSegmentCollection(ROUTE_SEGMENTS_MOCK);
+    const result = collection.getAllFixNamesInUse().sort();
+    const expectedResult = ['BAKRR', 'BESSY', 'FIXIX', 'HITME', 'JAKER', 'JESJI', 'MDDOG', 'MINEY', 'PIRMD', 'RBELL', 'ROPPR', 'TARRK', 'WASTE'];
+
+    t.true(_isEqual(result, expectedResult));
```
Comment (n8rzz) [inline test/navigationLibrary/StandardRoute/RouteSegmentCollection.spec.js]: I know we do this all over the place,  but remember you can use Ava native method `t.deepEqual()` here instead of `_isEqual()`

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js (hunk 14 above)
Comment (erikquinn) [reply src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: done

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js (hunk 15 above)
Comment (erikquinn) [reply src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: done

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js (hunk 17 above)
Comment (erikquinn) [reply src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: nah, that makes sense to me!

done

File: src/assets/scripts/client/navigationLibrary/NavigationLibrary.js (hunk 16 above)
Comment (erikquinn) [reply src/assets/scripts/client/navigationLibrary/NavigationLibrary.js]: ah yes! those static methods. good call, i just forgot about them.

done

File: src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteModel.js (hunk 18 above)
Comment (erikquinn) [reply src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteModel.js]: changed to:
```
const drawFixNames = this.draw.reduce((list, fix) => list.concat(fix))
    .map((fixName) => fixName.replace('*', ''));

const allFixNames = [
    ...this._entryCollection.getAllFixNamesInUse(),
    ...this._bodySegmentModel.getAllFixNamesInUse(),
    ...this._exitCollection.getAllFixNamesInUse(),
    ...drawFixNames
];
```

File: src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteWaypointModel.js (hunk 19 above)
Comment (erikquinn) [reply src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteWaypointModel.js]: I guess the point was that it sounds like the name of a method that would take an argument: `.clonePositionFromFix(fixName)`, when it actually does not. It just looks up and clones the position of the FixModel for which the StandardRouteWaypointModel is named. As I look again, it seems more okay now. Removed the note.

File: test/navigationLibrary/StandardRoute/RouteSegmentCollection.spec.js (hunk 20 above)
Comment (erikquinn) [reply test/navigationLibrary/StandardRoute/RouteSegmentCollection.spec.js]: :+1: done

File: src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteWaypointModel.js (hunk 19 above)
Comment (n8rzz) [reply src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteWaypointModel.js]: 🤔 thats a really good point.  That is certainly a better way to do that because it is a little handily at the moment.  This method is doing too much in that respect, it would be much easier to ask the `FixCollection` for a cloned `PositionModel` instead of _'Give me the fix.  Ok now Mr. Fix, give me a clone of your `PositionModel`.

Maybe add a `TODO` here to update that.  Good callout. 👍

**Comment by n8rzz** [comment]
@erikquinn be sure to fix that failing test before you merge, though 😉

**Comment by erikquinn** [comment]
:+1: all done!
